executor/examples/pre-execute-hook.sh

```
@@ -0,0 +1,89 @@
+    log "Creating target skills directory: $TARGET_SKILLS_DIR"
+    mkdir -p "$TARGET_SKILLS_DIR"
+    if [ $? -ne 0 ]; then
+        log "ERROR: Failed to create target skills directory"
+        exit 1
+    fi
+fi
+
+log "Copying skills from $SOURCE_SKILL_DIR to $TARGET_SKILLS_DIR/"
+cp -r "$SOURCE_SKILL_DIR" "$TARGET_SKILLS_DIR/"
+
+if [ $? -eq 0 ]; then
```

Shellcheck (0.11.0)

[style] Check exit code directly with e.g. 'if mycmd;', not indirectly with $?.

(SC2181)